src/Mailgun/Api/AbstractApi.php

```diff
@@ -39,6 +44,25 @@ public function __construct(HttpClient $httpClient, RequestFactory $requestFacto
     }
 
     /**
+     * Attempts to safely deserialize the response into the given class.
+     * If the HTTP return code != 200, deserializes into SimpleResponse::class
+     * to contain the error message and any other information provided.
+     *
+     * @param ResponseInterface $response
+     * @param string            $className
+     *
+     * @return $class|SimpleResponse
+     */
+    protected function safeDeserialize(ResponseInterface $response, $className)
```

Comment:
Hm, Im not sure about this. There are 3 possible outcomes of an API request. 

1) Success
2) [Common error code](https://documentation.mailgun.com/api-intro.html#errors)
3) Special HTTP status. Like 201 or 202. 

Number 2 should be handled in AbstractApi but 1 and 3 should be handled on each specific resource. 

👎  for this function.